project/AkkaDisciplinePlugin.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka
+
+import sbt._
+import Keys.{scalacOptions, _}
+import sbt.plugins.JvmPlugin
+
+/**
+  * Initial tests found:
+  * `akka-actor` 151 errors with `-Xfatal-warnings`, 6 without the flag
+  */
+object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
+
+  import scoverage.ScoverageKeys.{coverageFailOnMinimum, coverageHighlighting, coverageMinimum}
+  import scalafix.sbt.ScalafixPlugin
+
+  override def trigger: PluginTrigger = allRequirements
+  override def requires: Plugins = JvmPlugin && ScalafixPlugin
+
+  override lazy val projectSettings = disciplineSettings
+
+  lazy val scalaFixSettings = Seq(
+    Compile / scalacOptions += "-Yrangepos")
+
+  lazy val scoverageSettings = Seq(
+    coverageMinimum := 60, // TODO set during review, may need tuning per module as well
+    coverageFailOnMinimum := false,
+    coverageHighlighting := true)
```

Comment:
So this isn't enabled or failing anything currently?

Reply:
Actually I need to add logic around that for if Scala 2.11...